roadmap/implementers-guide/src/node/backing/prospective-parachains.md

```diff
@@ -0,0 +1,163 @@
+    relay-parent hash, which is a descendant of given candidate hashes.
+  - Sent by the Provisioner when requesting backable candidates, when
+    selecting candidates for a given relay-parent.
+- `ProspectiveParachainsMessage::GetHypotheticalFrontier`
+  - Gets the hypothetical frontier membership of candidates with the
+    given properties under the specified active leaves' fragment trees.
+  - Sent by the Backing Subsystem when sanity-checking whether a candidate can
+    be seconded based on its hypothetical frontiers.
+- `ProspectiveParachainsMessage::GetTreeMembership`
+  - Gets the membership of the candidate in all fragment trees.
+  - Sent by the Backing Subsystem when it needs to update the candidates
+    seconded at various depths under new active leaves.
+- `ProspectiveParachainsMessage::GetMinimumRelayParents`
+  - Gets the minimum accepted relay-parent number for each para in the
+    fragment tree for the given relay-chain block hash.
+  - That is, this returns the minimum relay-parent block number in the
+    same branch of the relay-chain which is accepted in the fragment
+    tree for each para-id.
+  - Sent by the Backing, Statement Distribution, and Collator Protocol
+    subsystems when activating leaves in the implicit view.
+- `ProspectiveParachainsMessage::GetProspectiveValidationData`
+  - Gets the validation data of some prospective candidate. The
+    candidate doesn't need to be part of any fragment tree.
+  - Sent by the Collator Protocol subsystem (validator side) when
+    handling a fetched collation result.
+
+### Outgoing
+
+- `RuntimeApiRequest::StagingParaBackingState`
+  - Gets the backing state of the given para (the constraints of the para and
+    candidates pending availability).
+- `RuntimeApiRequest::AvailabilityCores`
+  - Gets information on all availability cores.
+- `ChainApiMessage::Ancestors`
+  - Requests the `k` ancestor block hashes of a block with the given
+    hash.
+- `ChainApiMessage::BlockHeader`
+  - Requests the block header by hash.
+
+## Glossary
+
+- **Candidate storage:** Stores candidates and information about them
+  such as their relay-parents and their backing states. Is indexed in
+  various ways.
+- **Constraints:**
+  - Constraints on the actions that can be taken by a new parachain
+    block.
+  - Exhaustively define the set of valid inputs and outputs to parachain
+    execution.
+- **Fragment:** A prospective para block (that is, a block not yet referenced by
+  the relay-chain). Fragments are anchored to the relay-chain at a particular
+  relay-parent.
+- **Fragment tree:**
+  - A tree of fragments. Together, these fragments define one or more
+    prospective paths a parachain's state may transition through.
+  - See the "Fragment Tree" section.
+- **Inclusion emulation:** Emulation of the logic that the runtime uses
+  for checking parachain blocks.
+- **Prospective parachains:** Trees of potential (but likely) future
```

Comment:
Is there a way we could combine the definitions of `prospective parachains` and `fragment tree`? They seem to be addressing the same concept, though fragment tree addresses implementation a bit more.

Reply:
Good catch! I just removed this entry, since the whole page is about prospective parachains. 🙃